src/context-selection/period-selector-bar-item/use-periods.js

```diff
@@ -15,13 +15,16 @@ export default function usePeriods({
     year,
     dateLimit,
 }) {
+    const nowAtServerTimezone = useNowAtServerTimezone()
+    const dateString = nowAtServerTimezone.toISOString().split('T')[0]
```

Comment:
I'm not sure I get the use of the `.toISOString()` here?

For example, I'm in Auckland and it's 4:00 AM on 2024-01-01
My server is in Melbourne and it's 1:00 AM on 2024-01-01

So I send my nowAtServerTimeZone (which I've tricked to be 1:00 AM Auckland time on 2024-01-01) and now I get 11:00 AM on 2023-12-31 UTC from the conversion in toISOString(), which then causes my year limit to be 2023, even though both I and my server are in 2024?

I might also be misunderstanding. Either way, given this is only an issue on year boundaries, I don't think anyone will use the app on 31 December or 1 January, so maybe not worth it thinking it through anymore 🤪

Reply:
Very good point. I misused the `.toISOString()` method here, because I had a misconception of what it does. My assumption was that it would return a string representation of a local date. I didn't consult [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString), but if I had, I would have read:
> The timezone is always zero UTC offset, as denoted by the suffix `Z`

Instead of keeping this flawed logic in the codebase, I created a helper in 223d3ec which will create a date string from a local date.

Reply:
looks good, but I wonder if it might just be easier in the end to set the time portion to 0 and not convert back and forth to a string so as to avoid the exposure to parsing issues?
It seems like we're doing this back and forth string to date conversion (`const currentDate = new Date(dateString)`) in order to get a date with 00:00:00 time stamp, but we could also achieve that with:
```
nowAtServerTimezone.setHours(0)
nowAtServerTimezone.setMinutes(0)
nowAtServerTimezone.setSeconds(0)
nowAtServerTimezone.setMilliseconds(0)
```

Reply:
The conversion to a string is intentional, because we need to use it in the dependency array of the `useEffect` hook. `nowAtServerTimezone` is producing a new object reference on each render, so if we were to use it directly as a dependency here, the effect would run very often. The `dateString` on the other hand is a primitive, which will be recomputed on each render, but produce the same value unless the date actually changes. As such adding this variable to the `useEffect` dependency array will only trigger a re-render when the date actually changes, which is what we want.